src/pod/main.cpp

```diff
@@ -11,19 +16,27 @@ int main(int argc, char **argv)
   hyped::core::WallClock time;
   hyped::core::Timer timer(time);
   const auto execution_time = timer.measureExecutionTime([time]() {
-    hyped::core::Logger logger("GPIO", hyped::core::LogLevel::kDebug, time);
-    hyped::io::HardwareGpio gpio(logger);
-    auto gpio_reader_opt = gpio.getReader(0, hyped::io::Edge::kNone);
-    if (!gpio_reader_opt) {
-      logger.log(hyped::core::LogLevel::kFatal, "Error");
+    hyped::core::Logger logger("MQTT", hyped::core::LogLevel::kDebug, time);
+    const std::string id   = "test";
+    const uint16_t port    = 8080;
+    const std::string host = "localhost";
+    auto optional_mqtt     = hyped::core::Mqtt::create(logger, id, host, port);
+    if (!optional_mqtt) {
+      std::cout << "Failed to connect to MQTT broker" << std::endl;
       return;
     }
-    auto gpio_reader = *gpio_reader_opt;
-    if (gpio_reader->read() == hyped::core::DigitalSignal::kHigh) {
-      logger.log(hyped::core::LogLevel::kInfo, "High");
-    } else {
-      logger.log(hyped::core::LogLevel::kInfo, "Low");
+    auto mqtt        = *optional_mqtt;
+    const auto topic = hyped::core::MqttTopic::kTest;
+    rapidjson::Document message_payload;
+    message_payload.SetObject();
+    message_payload.AddMember("word", "Hello world!", message_payload.GetAllocator());
+    message_payload.AddMember("number", 42, message_payload.GetAllocator());
+    const auto message = hyped::core::MqttMessage{
+      .header  = {.timestamp = 0, .priority = hyped::core::MqttMessagePriority::kCritical},
+      .payload = message_payload,
     };
```

Comment:
Instead of initializing the message like this I used:
```
const hyped::core::MqttMessage message{topic, header, message_payload};
```
although there may be a better way to do this cause manually constructing the header and payload will be annoying every time we send a message.